Actions-and-issues.md

```diff
@@ -0,0 +1,51 @@
+## Table of contents
+
+* [Introduction](#introduction)
+* [Actions](#actions)
+* [Issues](#issues)
+* [Using actions and issues](#using-actions-and-issues)
+
+## Introduction
+
+As a user plays through an exploration, we want to track their progress to identify any problems. If we do find problems with the exploration, we can raise those to the exploration creator. For example, if a user gets stuck in a cycle of cards or submits many incorrect answers to an interaction, the exploration could probably be improved.
+
+We store the steps a user takes through an exploration as _actions_, and we create _issues_ to keep track of the problems that these actions indicate. Both actions and issues are defined as [[schemas|Schemas]].
+
+## Actions
+
+There are three actions defined at [`extensions/actions/`](https://github.com/oppia/oppia/tree/develop/extensions/actions):
+
+* `ExplorationStart`: This action is recorded when a learner starts an exploration. It stores the name of the state (i.e. card) they start at.
+
+* `AnswerSubmit`: This action is recorded when a learner submits a response to an interaction. It stores the ID of the interaction, the name of the state with the interaction, the answer that the learner submitted, the exploration's feedback for that answer, the next state they transition to, and how long the learner spent in the state. Note that the interaction ID is just its name, for example `ItemSelectionInput`.
+
+* `ExplorationQuit`: This action is recorded when a learner leaves an exploration. It stores the name of the state at which the learner quit and how long the user spent in the last state they visited.
+
+These actions are recorded using the functions provided by the [stats reporting service](https://github.com/oppia/oppia/tree/develop/core/templates/pages/exploration-player-page/services/stats-reporting.service.ts). For example, when a learner transitions into a terminal state (a card that ends the exploration), the [conversation skin directive](https://github.com/oppia/oppia/tree/develop/core/templates/pages/exploration-player-page/learner-experience/conversation-skin.directive.ts) calls `StatsReportingService.recordExplorationCompleted()`.
+
+When a learner's journey through an exploration goes smoothly, we don't store the actions anywhere; however, when we detect a problem in a learner's journey, then we create an issue that stores the associated actions.
```

Comment:
Maybe add, somewhere: "Note that we do not store any details about the learner's identity."

Reply:
done